test/common/config/subscription_impl_test.cc

```diff
@@ -179,9 +179,9 @@ TEST_P(SubscriptionImplInitFetchTimeoutTest, InitialFetchTimeout) {
   expectEnableInitFetchTimeoutTimer(std::chrono::milliseconds(1000));
   startSubscription({"cluster0", "cluster1"});
   EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
-  if (GetParam() == SubscriptionType::Http) {
-    expectDisableInitFetchTimeoutTimer();
-  }
+
+  expectDisableInitFetchTimeoutTimer();
+
```

Comment:
Can you add a test that validates the stats are updated?

Reply:
It is already verified at the end of the test   EXPECT_TRUE(statsAre(1, 0, 0, 0, 1, 0, 0, ""));

Reply:
Why was it passing before if that is validating the regression?

Reply:
see the implementation of ` expectConfigUpdateFailed()` before these changes (here: https://github.com/envoyproxy/envoy/pull/15114/files#diff-c09580e89a4c15ba79b2473fd1ae255b463fbde941649cb2420408b0261705a5L165).

Reply:
Ah, I see. Thanks for pointing this out.